File: apps/api/app/schemas/features.py
from typing import Literal

from pydantic import BaseModel


class FeatureAccessResponse(BaseModel):
    plan_name: Literal[
        "free",
        "pro",
        "family",
    ]

    max_chores: int | None
    unlimited_chores: bool

    advanced_reminders: bool
    analytics: bool
    household_sharing: bool
    ai_planning: bool

File: apps/web/src/types/features.ts
export type FeaturePlan =
  | "free"
  | "pro"
  | "family";

export type FeatureAccess = {
  plan_name: FeaturePlan;
  max_chores: number | null;
  unlimited_chores: boolean;
  advanced_reminders: boolean;
  analytics: boolean;
  household_sharing: boolean;
  ai_planning: boolean;
};